tsl/src/continuous_aggs/repair.c

```diff
@@ -410,12 +356,37 @@ continuous_agg_get_bucket_function(PG_FUNCTION_ARGS)
 	Assert(direct_query != NULL);
 	Assert(direct_query->commandType == CMD_SELECT);
 
-	/* Process the query and collect function information */
-	CaggQueryWalkerContext context = { 0 };
-	context.mat_hypertable_id = mat_hypertable_id;
-	context.bucket_fuction = InvalidOid;
+	ListCell *l;
+	bool found = false;
+	foreach (l, direct_query->groupClause)
+	{
+		SortGroupClause *sgc = (SortGroupClause *) lfirst(l);
+		TargetEntry *tle = get_sortgroupclause_tle(sgc, direct_query->targetList);
+
+		if (IsA(tle->expr, FuncExpr))
+		{
+			FuncExpr *fe = ((FuncExpr *) tle->expr);
+
+			/* Filter any non bucketing functions */
+			FuncInfo *finfo = ts_func_cache_get_bucketing_func(fe->funcid);
+			if (finfo == NULL || !finfo->is_bucketing_func)
```

Comment:
Given that you call `ts_func_get_bucketing_func()` I believe it can never return a func with `is_bucketing_func` set to false (if such a func even exists in the cache). Therefore, either remove or assert:

```suggestion
			if (finfo == NULL)
```

Reply:
Make sense, removed the check and added an assert after the NULL check.